src/main/cpp/src/decimal_utils.cu

```diff
@@ -766,7 +774,7 @@ struct dec128_divider : public thrust::unary_function<cudf::size_type, __int128_
 
       auto const result = divide_and_round(n, d);
 
-      overflows[i] = !result.fits_in_128_bits();
+      overflows[i] = is_greater_than_decimal_38(result);
```

Comment:
We explicitly had multiply and divide check that it fits inside of a decimal128, not that it fits in a precision of 38, because of some odd corner cases that are possible with spark versions prior to 3.4. I don't remember off of the top of my head exactly what it was, or if it was just because the CheckOverflow would run afterwards and fix it there. I am going to have to think about this more to be sure that it is not going to mess something up for the older versions of Spark.

Reply:
Talked to @jlowe and I think this is fine. I was remembering something related to the SUM aggregation.